.github/workflows/ci-build.yml

```diff
@@ -256,10 +256,25 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [windows-latest, windows-2019]
-        arch: [x86, x64]
-
-    name: Windows (${{ matrix.os }} / ${{ matrix.arch }})
+        include:
+        - name: windows-latest / x86
+          arch: x86
+          os: windows-latest
+        - name: windows-latest / x64
+          arch: x64
+          os: windows-latest
+        - name: windows-2019 / x86
+          arch: x86
+          os: windows-2019
+        - name: windows-2019 / x64
```

Comment:
personally, I wouldn't do this `name` entry. Without it, and having the `name: Windows (..)` as it was, saves a field we can make mistakes with.

Reply:
owh, no, I see why you did this. Maybe we are better off adding this to the GCC Dedicated run. It is already a special run to check these kinds of things, and saves us from this trouble?

Reply:
Though how would you see the difference between the one with and the one without precompiled headers if it's only showing arch and OS?

Reply:
> owh, no, I see why you did this. Maybe we are better off adding this to the GCC Dedicated run. It is already a special run to check these kinds of things, and saves us from this trouble?

The reason I choose MSVC is because GCC, and Clang on Linux, will happily accept it in our current codebase to not include `optional` when using `std::optional`, whereas MSVC just fails to build. That's the whole reason I went for MSVC, and the x86 variant is because that's most picky about `size_t` to `int` conversions.

Though indeed making GCC Dedicate the one without precompiled headers is the easiest, but will it catch most of the issues? That's the conundrum I've been thinking about.

Reply:
I could argue that nobody is going to run MSVC without precompiled headers (as it is on by default on all new MSVC projects anyway), while I am sure there will be some Linux variants where it doesn't work (looking at you GCC 11) :P

I would go for our sanity, and keep the CI file as simple as we can .. it is already scary complex :)